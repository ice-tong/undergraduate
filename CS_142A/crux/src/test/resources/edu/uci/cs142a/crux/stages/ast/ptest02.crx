// test variable containing keyword

func main() : void {
  var a: bool;
  var notA: bool;
  let a = true;
  let notA = not a;
  ::printInt(notA);
}
