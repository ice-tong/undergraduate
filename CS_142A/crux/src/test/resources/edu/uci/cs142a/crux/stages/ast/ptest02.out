ast.DeclarationList(3,0)
  ast.FunctionDefinition(3,0)[Symbol(main), []]
    ast.StatementList(4,2)
      ast.VariableDeclaration(4,2)[Symbol(a)]
      ast.VariableDeclaration(5,2)[Symbol(notA)]
      ast.Assignment(6,2)
        ast.AddressOf(6,6)[a]
        ast.LiteralBool(6,10)[TRUE]
      ast.Assignment(7,2)
        ast.AddressOf(7,6)[notA]
        ast.LogicalNot(7,13)
          ast.Dereference(7,17)
            ast.AddressOf(7,17)[a]
      ast.Call(8,2)[Symbol(printInt)]
        ast.ExpressionList(8,13)
          ast.Dereference(8,13)
            ast.AddressOf(8,13)[notA]
